File: js/anagram2.js
exports.anagramsFor = function(word, listOfWords) {
    let searchWord = removeWhiteSpace(sort(word.split('')))
    let result = []
    for(let word in listOfWords) {
        if(JSON.stringify(removeWhiteSpace(sort(turnToArr(listOfWords[word])))) == JSON.stringify(searchWord)) {
            result.push(listOfWords[word])
        }
    }
    return result
};

function sort(x){
    return x.sort()
}

function removeWhiteSpace(x){
    return x.filter(word => word !== ' ');
}

function turnToArr(x){
    return x.split('')
}

File: python/anagram2.py
# Don't forget to run the tests (and create some of your own)

def anagrams_for(word, list_of_words):
	result = []
	word_sorted_list = sorting_words(list(word))
	for words in list_of_words:
		if sorting_words(list(words)) == word_sorted_list:
			result.append(words)	
	
	return result

def sorting_words(x):
	return sorted(x)

#create test cases to test for whitespace
def remove_whitespace(x):
	return filter(lambda char: char == ' ', x)